connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -0,0 +1,511 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.{lang => jl}
+import java.util.Locale
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.connect.proto.Expression.{Literal => GLiteral}
+import org.apache.spark.connect.proto.{NAReplace, Relation}
+import org.apache.spark.connect.proto.NAReplace.Replacement
+
+/**
+ * Functionality for working with missing data in `DataFrame`s.
+ *
+ * @since 3.4.0
+ */
+final class DataFrameNaFunctions private[sql](sparkSession: SparkSession, root: Relation) {
+
+  /**
+   * Returns a new `DataFrame` that drops rows containing any null or NaN values.
+   *
+   * @since 3.4.0
+   */
+  def drop(): DataFrame = {
+    sparkSession.newDataFrame { builder =>
```

Comment:
Is there a way we can reduce duplication. For example:
```scala
private def buildDropDataFrame(cols: Seq[String], minNonNulls: Option[Int]): DataFrame = {
  sparkSession.newDataFrame { builder =>
    val dropNaBuilder = builder.getDropNaBuilder.setInput(root).addAllCols(cols.asJava)
    minNonNulls.foreach(builder.setMinNonNulls)
}
```

Reply:
The same applies for fill.

Reply:
Done